homeassistant/components/esphome/climate.py

```diff
@@ -81,11 +88,12 @@ async def async_setup_entry(hass, entry, async_add_entities):
 def _climate_modes():
     return {
         ClimateMode.OFF: HVAC_MODE_OFF,
-        ClimateMode.AUTO: HVAC_MODE_HEAT_COOL,
+        ClimateMode.AUTO: HVAC_MODE_AUTO,
```

Comment:
Is the user able to change temperature in the auto mode? We have rules for what modes should be used.

https://github.com/home-assistant/core/blob/e0cd7072d6480ff314e4cf1935decc9a22d46a9e/homeassistant/components/climate/const.py#L12-L17

Reply:
In esphome, there was no `AUTO` mode, and `AUTO` was mapped to `HEAT_COO`. Now that we do have that it is properly mapped here a well. @OttoWinter can comment more.

Reply:
> Is the user able to change temperature in the auto mode?

Someone familiar with esphome and its auto mode needs to answer my question for the review to progress.

Reply:
@OttoWinter @jesserockz @glmnet Can any of you comment on this.

Reply:
I guess this would need to be a soft rule on esphome as it is in home assistant. The esphome climate component is just an abstract base class for a number of devices specific implementations, so I guess those will need to ensure that AUTO will only be mapped to device states that do not allow manual setpoint changes.

Reply:
@MartinHjelmare I have just been "fixing" the esphome codebase after a few forgotten things around esphome/esphome#1471 and came to understand what the difference between AUTO and HEAT_COOL was.

Currently there are now no components in ESPHome that use the AUTO mode. But until the HA side is merged they will not be able to be developed as the HA connection side crashes when connecting to an ESPHome node with supported mode AUTO.